### PR TITLE
[CLNP-5022] feat: add message input related eventHandlers

### PR DESCRIPTION
**Pull request by AhyoungRyu**
Addresses https://sendbird.atlassian.net/browse/CLNP-5022

### Main Changes 
```
eventHandlers?: {
 ...
  message?: {
    onSendMessageFailed?: (message: CoreMessageType, error: unknown) => void;
    onUpdateMessageFailed?: (message: CoreMessageType, error: unknown) => void;
    onFileUploadFailed?: (error: unknown) => void;
    // any message related event handlers can be added here
  }
}
```
The original request is only about sending message related event, but just added two more (onUpdateMessageFailed / onFileUploadFailed)

### Checklist

Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If unsure, ask the members.
This is a reminder of what we look for before merging your code.

- [x] **All tests pass locally with my changes**
- [x] **I have added tests that prove my fix is effective or that my feature works**
- [ ] **Public components / utils / props are appropriately exported**
- [ ] I have added necessary documentation (if appropriate)

### CODE REVIEW

File: src/ui/MessageInput/index.tsx [hunk 1]
```diff
@@ -391,38 +391,45 @@ const MessageInput = React.forwardRef<HTMLInputElement, MessageInputProps>((prop
   }, [isMentionEnabled]);
 
   const sendMessage = () => {
-    const textField = internalRef?.current;
-    if (!isEdit && textField?.textContent) {
-      const { messageText, mentionTemplate } = extractTextAndMentions(textField.childNodes);
-      const params = { message: messageText, mentionTemplate };
-      onSendMessage(params);
-      resetInput(internalRef);
+    try {
```
Comment (AhyoungRyu) [inline src/ui/MessageInput/index.tsx:394]: Just wrapped with try - catch.

File: src/ui/MessageInput/index.tsx [hunk 2]
```diff
@@ -391,38 +391,45 @@ const MessageInput = React.forwardRef<HTMLInputElement, MessageInputProps>((prop
...
   const isEditDisabled = !internalRef?.current?.textContent?.trim();
   const editMessage = () => {
-    const textField = internalRef?.current;
-    const messageId = message?.messageId;
-    if (isEdit && messageId && textField) {
-      const { messageText, mentionTemplate } = extractTextAndMentions(textField.childNodes);
-      const params = { messageId, message: messageText, mentionTemplate };
-      onUpdateMessage(params);
-      resetInput(internalRef);
+    try {
```
Comment (AhyoungRyu) [inline src/ui/MessageInput/index.tsx:422]: Same change

File: src/ui/MessageInput/index.tsx [hunk 3]
```diff
@@ -433,6 +440,19 @@ const MessageInput = React.forwardRef<HTMLInputElement, MessageInputProps>((prop
     setHeight,
   });
 
+  const uploadFile = (event: React.ChangeEvent<HTMLInputElement>) => {
+    const { files } = event.currentTarget;
+    try {
+      if (files) {
+        onFileUpload(files && files.length === 1 ? [files[0]] : Array.from(files));
```
Comment (chrisallo) [inline src/ui/MessageInput/index.tsx:447]: 요거 전부터 있긴 코드이긴 한데 궁금한게 있어서ㅎㅎ `onFileUpload(Array.from(files))` 이렇게 바로 해버리는거랑 현재 코드랑 동작의 차이가 있을까요?

**Review by chrisallo — APPROVED** [review]
lgtm!

File: src/ui/MessageInput/index.tsx (hunk 3 above)
Comment (AhyoungRyu) [reply src/ui/MessageInput/index.tsx:447]: Oh right. I didn't realize it.

Comment (AhyoungRyu) [reply src/ui/MessageInput/index.tsx:447]:
```suggestion
        onFileUpload(Array.from(files));
```

**Pull request merged**